doc/src/advanced_tutorial/add_move.md

```diff
@@ -0,0 +1,70 @@
+
+To follow this tutorial, you should know about the following basic concepts of rust:
+- structs
+- Traits
+- functions
+
+Ready? Let's start.
+
+## Where are source files for the MC propagator located?
+
+Let's start by taking a look at where MC related source files are located within the Lumol project. Navigate to the MC folder, which you will find at `lumol/src/core/src/sim/mc`.
+
+The folders' structure looks like so:
+```
+$ tree .
+.
+├── mod.rs
+├── monte_carlo.rs
+└── moves
+    ├── mod.rs
+    ├── rotate.rs
+    :
+    └── translate.rs
+```
+
+Initially, all of our work will take place within the `moves` directory.
+
+In rust terms, a move is a struct that implements the `MCMove` trait. The trait is defined in the `moves/mod.rs` file. Let's have a look:
+
+```rust
+// File: move/mod.rs
+pub trait MCMove {
+    fn describe(&self) -> &str;
+    fn setup(&mut self, system: &System);
+    fn prepare(&mut self, system: &mut System, rng: &mut Box<Rng>) -> bool;
+    fn cost(&self, system: &System, beta: f64, cache: &mut EnergyCache) -> f64;
+    fn apply(&mut self, system: &mut System);
+    fn restore(&mut self, system: &mut System);
+    fn update_amplitude(&mut self, scaling_factor: Option<f64>);
+}
+```
+
+In Lumol, MC propagation (which is defined in `monte-carlo.rs`) is divided into several steps.
+Basically, each of these steps makes use of one of the above shown methods of `MCMove`.
+As you can see the `System` structure is involved in most functions.
+The `System` is the heart of Lumol and is a quite sophisticated structure.
+At the moment it is sufficient for us to notice that the system contains the positions (`system.positions`) and that we can access the energy of the current configuration directly (`system.total_energy()`).
+
+Going through the functions one-by-one:
+
+- The `describe` function returns a `&str` with a brief description of the move, e.g. "translation of a molecule" or "rotation of a molecule". Primarily, this function is used for log files and errors.
+- `setup` is called *once before the actual simulation is run* for every move. You can use this function to set `System` dependent parameter.
+- `prepare` can be used to perform small computations to set up the moves' parameter that depend on the state of the `System`.
+- The `cost` function is the core functionality of a move. It is the function where most (if not all) of the computational work takes place. The return value is used from the propagator to decide if a proposed move is accepted or rejected. 
+- If a move is accepted, we update the `System` by invoking the `apply` function.
+- If a move is rejected, we restore the `System` to its old state by invoking the `restore` function (if we modified it beforehand).
+- The `update_amplitude` function can be used to increase the efficiency of a simulation run by adjusting how much a move actually changes the system.
```

Comment:
Maybe adding a note / link to an explanation for this, and saying not to worry about it in the simplest cases?